dispatcher/MongoConnect.py

```diff
@@ -353,8 +364,8 @@ def get_super_detector(self):
         - case E: tpc unlinked, mv and nv linked
         We will check the compatibility of the linked mode for a pair of detectors per time.
         """
-        ret = {'tpc': {'controller': list(self.dc['tpc']['controller'].keys()),
-                       'readers': list(self.dc['tpc']['readers'].keys()),
+        ret = {'tpc': {'controller': self.dc['tpc']['controller'][:],
```

Comment:
I thought you wanted the `list` to make sure you are keeping a copy rather than a view, does `[:]` achieve it cleaner?

Reply:
So for one, `self.dc['tpc']['controller']` wasn't a dict, it's a list, so `.keys()` doesn't work. A full `[:]` slice makes a copy, so we can modify it without affecting the master config.
```
>>> x = ['a']
>>> y = x
>>> z = x[:]
>>> y.append('b')
>>> x, y, z
(['a', 'b'], ['a', 'b'], ['a'])
```